tasks.py

```diff
@@ -215,47 +215,92 @@ def load_osm_streets(ctx, files=[]):
-        additional_params += _get_cli_param(conf.get("nb_replicas"), "--nb-replicas")
-        run_rust_binary(
-            ctx,
-            "mimir",
-            "bano2mimir",
-            files,
-            "--input {ctx.addresses.bano.file} \
-            --connection-string {ctx.es} \
-            {additional_params} \
-            --dataset {ctx.dataset}".format(
-                ctx=ctx, additional_params=additional_params
-            ),
-        )
-    if addr_config.get("oa", {}).get("path"):
-        logging.info("importing oa addresses")
+    if not ctx.addresses.get("use_deduplicator", False):
+        addr_config = ctx.get("addresses")
+        if addr_config.get("bano", {}).get("file"):
+            logging.info("importing bano addresses")
+            conf = ctx.addresses.bano
+            additional_params = _get_cli_param(conf.get("nb_threads"), "--nb-threads")
+            additional_params += _get_cli_param(conf.get("nb_shards"), "--nb-shards")
+            additional_params += _get_cli_param(conf.get("nb_replicas"), "--nb-replicas")
+            run_rust_binary(
+                ctx,
+                "mimir",
+                "bano2mimir",
+                files,
+                "--input {ctx.addresses.bano.file} \
+                --connection-string {ctx.es} \
+                {additional_params} \
+                --dataset {ctx.dataset}".format(
+                    ctx=ctx, additional_params=additional_params
+                ),
+            )
+        if addr_config.get("oa", {}).get("path"):
+            logging.info("importing oa addresses")
+            conf = ctx.addresses.oa
+            additional_params = _get_cli_param(conf.get("nb_threads"), "--nb-threads")
+            additional_params += _get_cli_param(conf.get("nb_shards"), "--nb-shards")
+            additional_params += _get_cli_param(conf.get("nb_replicas"), "--nb-replicas")
+            run_rust_binary(
+                ctx,
+                "mimir",
+                "openaddresses2mimir",
+                files,
+                "--input {ctx.addresses.oa.path} \
+                --connection-string {ctx.es} \
+                {additional_params} \
+                --dataset {ctx.dataset}".format(
+                    ctx=ctx, additional_params=additional_params
+                ),
+            )
+        return
+
+    output_csv = "output.csv.gz"
+
+    options = []
+    additional_params = ''
+    if ctx.addresses.get("bano", {}).get("file"):
+        options.append("--bano")
+        options.append(ctx.addresses.bano.file)
+    if ctx.addresses.get("oa", {}).get("file"):
+        options.append("--openaddresses")
+        options.append(ctx.addresses.oa.file)
         conf = ctx.addresses.oa
         additional_params = _get_cli_param(conf.get("nb_threads"), "--nb-threads")
         additional_params += _get_cli_param(conf.get("nb_shards"), "--nb-shards")
         additional_params += _get_cli_param(conf.get("nb_replicas"), "--nb-replicas")
```

Comment:
These "additional_params"  should be defined, even if no `oa.file` is used during the deduplication.